catt/controllers.py

```diff
@@ -205,12 +205,18 @@ def get_data(self, name):
 
 
 class CastState(CattStore):
-    def __init__(self, state_path, create_dir=False):
+    def __init__(self, state_path, mode="read"):
```

Comment:
Hmm, why is `mode` here a string rather than an `enum` or something similar? A string seems pretty error-prone.

Reply:
Something like?:
```
class Mode(enum.Enum):
    READ = auto()
    CONF = auto()
    ARBI = auto()
```
Forgive my ignorance, but why would the string be error-prone?

Reply:
The way you've implemented it, it'll raise an error, which is good, but I've seen this be implemented as "if x: elif y: else:", and then people misspelled "conf" and boom, the mode was "read". It's also less semantic (`def myfunc(mode: str)` is much less readable than `def myfunc(mode: StateMode)`) and the type checker can't check it as easily.

Reply:
Thank you for the explanation. Did you like my enum example?

Reply:
Sorry, yes, that's perfect. I would name it something more descriptive than `Mode` ("what kind of mode?"), but other than that it's great.

Reply:
I really wish they had something like `MyEnum = enum_generator(["foo", "bar", "baz"])` instead of the heavier class-based approach, but oh well.